### PR TITLE
Set of patches for core

### PR DESCRIPTION
**Pull request by micmarty**
1. Allow adding new hosts in runtime.
2. Replace printing with logging + add logging config.
3. Remove ThreadedManager (duplicate of StoppableThread) and clean up inheritance boilerplate.
4. Introduce a minimal interval value between monitoring actions.
5. Autopep8 on files
Tested on clean env with python 3.5

### CODE REVIEW

File: tensorhive/core/managers/TensorHiveManager.py [hunk 1]
```diff
@@ -1,55 +1,40 @@
...
+class TensorHiveManager(StoppableThread, metaclass=Singleton):
     """Heart of the whole engine"""
 
     def __init__(self):
         super().__init__()
         self.infrastructure_manager = InfrastructureManager()
-
-        print('{bg_red}WARNING! You need to replace hostnames and usernames to your own in tensorhive/config.py{style_reset}'.format(bg_red=Back.RED,style_reset=Style.RESET_ALL))
-
+        log.warning('You need to replace hostnames and usernames to your own in tensorhive/config.py')
```
Comment (roscisz) [inline tensorhive/core/managers/TensorHiveManager.py]: This warning suggests that I forgot to configure something. Maybe it should be INFO level or show warning only if the usernames are not configured?

File: tensorhive/api/APIServer.py [hunk 2]
```diff
@@ -2,29 +2,26 @@
...
 
 class APIServer():
     def start(self):
-        # TODO Enable debug mode
-        # TODO Read settings from config
-        # FIXME: set log level for the whole app in cli
-        logging.basicConfig(level=logging.INFO)
         app = connexion.FlaskApp(__name__)
+        log.info(
+            'API docs (Swagger UI) available at: http://<address>:<port>/v1.0/ui/')
```
Comment (roscisz) [inline tensorhive/api/APIServer.py]: Is it intentional, that address and port are not given explicitly? I would expect a working URL.

File: tensorhive/config.py [hunk 3]
```diff
@@ -28,17 +35,32 @@ class SSHConfig():
     '''
     AVAILABLE_NODES = {
```
Comment (roscisz) [inline tensorhive/config.py]: I think we should get rid of the specific user credentials in the code, just leave a placeholder like:
AVAILABLE_NODES = {
#'<hostname>': {'<username>': '<password>'}
}
and then maybe detect empty AVAILABLE_NODES in THManager?

File: tensorhive/api/APIServer.py (hunk 2 above)
Comment (micmarty) [reply tensorhive/api/APIServer.py]: Fixed: 5b907841ad4fb6cf03bfa7acdb7ad0bfa126a0e1

File: tensorhive/core/managers/TensorHiveManager.py (hunk 1 above)
Comment (micmarty) [reply tensorhive/core/managers/TensorHiveManager.py]: Fixed: 154a6ac2df349ee2c584e6decaceb3386fa7e4f5
Good enough?

File: tensorhive/config.py (hunk 3 above)
Comment (micmarty) [reply tensorhive/config.py]: Fixed credentials: 9b860728fcb0d884fb581f007210b7a36294d611
Fixed empty nodes dict: 154a6ac2df349ee2c584e6decaceb3386fa7e4f5

**Comment by micmarty** [comment]
If there would be no more change requests, I'm able to resolve the conflicts.
**Waiting for review**. @roscisz @tomenet @filschod

**Pull request merged**